### PR TITLE
Fixes #724

### LINKED ISSUE #724: Uncaught ReferenceError: MutationEvent is not defined
# Bug Report

## Prerequisites

- [x] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [x] Are you running the latest version?
- [x] Did you perform a search in the issues?

## Environment Details and Version

* Chrome version: 130.0.6723.92 (Official Build) (64-bit)
* Operating System: Windows 10

## Description

I'm experiencing an error when using the Piral Blazor converter with my application. The JavaScript console of my Chrome browser (version 130.0.6723.92) shows a repeated exception:

```
Uncaught ReferenceError: MutationEvent is not defined
    at n.Vht (events.js:61:28)
    at s.invokeTask (zone.js:402:33)
    at i.runTask (zone.js:159:47)
    at c.invokeTask [as invoke] (zone.js:483:34)
    at y (zone.js:1138:18)
    at w (zone.js:1169:29)
    at n.x (zone.js:1202:16)
```

This error occurs over 763 times and even more frequently when I move the mouse around.

## Steps to Reproduce

1. Run your application using the Piral Blazor converter.
2. Open the JavaScript console in Chrome (version 130.0.6723.92).
3. Observe the repeated exception messages with `Uncaught ReferenceError: MutationEvent is not defined`.

## Expected behavior

I expect to see no errors or exceptions when running my application using the Piral Blazor converter.

## Actual behavior

There is a huge amount of errors slowing my application down.

## Possible Origin/Solution

Remove the check as the `MutationEvent` is no longer supported by modern browsers.

### PR DESCRIPTION
**Pull request by dheid**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

See https://github.com/smapiot/piral/issues/724

### CODE REVIEW

File: src/converters/piral-blazor/src/events.ts [hunk 1]
```diff
@@ -67,8 +67,7 @@ function dispatchToRoot(event: any) {
     performInternalNavigation(event);
   }
 
-  // the mutation event cannot be cloned (at least in Webkit-based browsers)
-  if (!(event instanceof MutationEvent) && !event.processed) {
```
Comment (FlorianRappl) [inline src/converters/piral-blazor/src/events.ts:71]: I think in general this change is fine, but I'd make it such that it won't impact browsers that still supply it (which would still run in the other issue; that a mutationevent wants to be cloned).

--> My proposal

```js
if ((typeof MutationEvent === 'undefined' || !(event instanceof MutationEvent)) && !event.processed) {
  // ...
}
```

**Review by FlorianRappl — COMMENTED** [review]
LGTM - maybe also consider backwards compatibility to avoid running into the original problem in other / older browsers.

**Comment by dheid** [comment]
@FlorianRappl I agreed multiple times to the contributor license, but it still says "Contributor License Agreement is not signed yet.". Do you have an idea, what I can do?

**Comment by FlorianRappl** [comment]
> @FlorianRappl I agreed multiple times to the contributor license, but it still says "Contributor License Agreement is not signed yet.". Do you have an idea, what I can do?

It either may be a problem with the bot or with the settings. Keep in mind that the bot is operating on your GitHub account, so it only has access to your registered email addresses. To verify that the commits are done by you the email you entered in your local git instance is compared to the email addresses of your GitHub account. Is the email matching? If not - you may need to enter that email address in your alias list.

**Comment by dheid** [comment]
Nice! That solved the license agreement issue. Thanks!

**Comment by FlorianRappl** [comment]
Awesome - thanks again for your contribution!

There are a couple of things that should enter in 1.8.0, but in general the version should come out this week.

**Comment by dheid** [comment]
Thank you so much for accepting this PR, merging it and for the release! This bugfix will help a lot of people, I suppose. It was a really nice experience contributing to Piral. Thx!

**Pull request merged**